components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -651,6 +644,73 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
         sendResponse(request, response, context);
     }
 
+    private void storeFedAuthSessionMapping(String sessionContextKey, AuthHistory authHistory)
+            throws UserSessionException {
+
+        UserSessionStore userSessionStore = UserSessionStore.getInstance();
+        if (!userSessionStore.hasExistingFederatedAuthSession(
+                authHistory.getIdpSessionIndex())) {
+            userSessionStore.storeFederatedAuthSessionInfo(sessionContextKey, authHistory);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Federated auth session with the id: %s already " +
+                        "exists.", authHistory.getIdpSessionIndex()));
+            }
```

Comment:
Is there any need to use String.format given that only one param is passed? We can simply concat

Reply:
Fixed with [f22af54](https://github.com/wso2/carbon-identity-framework/pull/4558/commits/f22af54d48148c23a43fda73c6b52870762a599c)